dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java

```diff
@@ -173,6 +173,17 @@ public Service initializeService(String repositoryId) {
     public abstract Workflow setupWorkflowVersions(String repositoryId, Workflow workflow, Optional<Workflow> existingWorkflow,
             Map<String, WorkflowVersion> existingDefaults);
 
+    /**
+     * Creates a basic workflow object with default values
+     * @param repository repository organization and name (ex. dockstore/dockstore-ui2)
+     * @return basic workflow object
+     */
+    public Workflow createStubBioworkflow(String repository) {
+        Workflow workflow = initializeWorkflow(repository, new BioWorkflow());
+        workflow.setDescriptorType(DescriptorLanguage.CWL);
```

Comment:
In WorkflowResource you test if the return value of this method can be null, but this line assumes it never can be.

BTW, I am a recent convert/zealot to almost never returning null and returning Optional instead, so if you think it can return null, I would suggest it return `Optional<Workflow>`